packages/pangea-sdk/pangea/services/authz.py

```diff
@@ -0,0 +1,344 @@
+        return str(self.value)
+
+    def __repr__(self):
+        return str(self.value)
+
+
+class Resource(PangeaResponseResult):
+    namespace: str
+    id: Optional[str] = None
+
+
+class Subject(PangeaResponseResult):
+    namespace: str
+    id: Optional[str] = None
+    action: Optional[str] = None
+
+
+class Tuple(PangeaResponseResult):
+    resource: Resource
+    relation: str
+    subject: Subject
+
+
+class TupleCreateRequest(APIRequestModel):
+    tuples: List[Tuple]
+
+
+class TupleCreateResult(PangeaResponseResult):
+    pass
+
+
+class TupleListFilter(APIRequestModel):
+    resource_namespace: Optional[str] = None
+    resource_namespace__contains: Optional[List[str]] = None
+    resource_namespace__in: Optional[List[str]] = None
+    resource_id: Optional[str] = None
+    resource_id__contains: Optional[List[str]] = None
+    resource_id__in: Optional[List[str]] = None
+    relation: Optional[str] = None
+    relation__contains: Optional[List[str]] = None
+    relation__in: Optional[List[str]] = None
+    subject_namespace: Optional[str] = None
+    subject_namespace__contains: Optional[List[str]] = None
+    subject_namespace__in: Optional[List[str]] = None
+    subject_id: Optional[str] = None
+    subject_id__contains: Optional[List[str]] = None
+    subject_id__in: Optional[List[str]] = None
+    subject_action: Optional[str] = None
+    subject_action__contains: Optional[List[str]] = None
+    subject_action__in: Optional[List[str]] = None
+
+
+class TupleListRequest(APIRequestModel):
+    filter: Optional[Union[Dict, TupleListFilter]] = None
+    size: Optional[int] = None
+    last: Optional[str] = None
+    order: Optional[ItemOrder] = None
+    order_by: Optional[TupleOrderBy] = None
+
+
+class TupleListResult(PangeaResponseResult):
+    tuples: List[Tuple]
+    last: str
+    count: int
+
+
+class TupleDeleteRequest(APIRequestModel):
+    tuples: List[Tuple]
+
+
+class TupleDeleteResult(PangeaResponseResult):
+    pass
+
+
+class CheckRequest(APIRequestModel):
+    resource: Resource
+    action: str
+    subject: Subject
+    debug: Optional[bool] = None
+    attributes: Optional[Dict[str, Any]] = None
+
+
+class DebugPath(APIResponseModel):
+    namespace: str
+    id: str
+    action: Optional[str]
+
+
+class Debug(APIResponseModel):
+    path: List[DebugPath]
+
+
+class CheckResult(PangeaResponseResult):
+    schema_id: str
+    schema_version: int
+    depth: int
+    allowed: bool
+    debug: Optional[Debug] = None
+
+
+class ListResourcesRequest(APIRequestModel):
+    namespace: str
+    action: str
+    subject: Subject
+
+
+class ListResourcesResult(PangeaResponseResult):
+    ids: List[str]
+
+
+class ListSubjectsRequest(APIRequestModel):
+    resource: Resource
+    action: str
+
+
+class ListSubjectsResult(PangeaResponseResult):
+    subjects: List[Subject]
+
+
+class AuthZ(ServiceBase):
+    """AuthZ service client.
+
+    Provides methods to interact with the Pangea AuthZ Service.
```

Comment:
We need a note to say that this is in Beta and subject to change.

Reply:
Done.